### PR TITLE
metadata parser

### PR DESCRIPTION
**Pull request by vratsel**

### CODE REVIEW

File: embedded/build.gradle [hunk 1]
```diff
@@ -26,8 +29,186 @@ dependencies {
...
+
+  ant.metadataParser(verbose:"false", classpathRef: 'metadata') {
+    path(pathToApi: "${metadataSrcDir}/java",
+         pathToImpl: "${metadataSrcDir}/java",
+         pathToServices: "${metadataSrcDir}/resources/META-INF/services")
+    descriptors() {
+      descriptor(pathToXsd: project(':common').file("src/main/resources/dtd/connector_1_0.dtd"),
+                 nameSpace: "j2ee",
+                 packageApi: "org.ironjacamar.shrinkwrap.descriptor.api.connector10",
+                 packageImpl:"org.ironjacamar.shrinkwrap.descriptor.impl.connector10",
```
Comment (jesperpedersen) [inline embedded/build.gradle]: These need to stay at "org.jboss.shrinkwrap" in order to align with the official JARs

File: embedded/build.gradle [hunk 2]
```diff
@@ -26,8 +29,186 @@ dependencies {
...
+
+  ant.metadataParser(verbose:"false", classpathRef: 'metadata') {
+    path(pathToApi: "${metadataSrcDir}/java",
+         pathToImpl: "${metadataSrcDir}/java",
+         pathToServices: "${metadataSrcDir}/resources/META-INF/services")
+    descriptors() {
+      descriptor(pathToXsd: project(':common').file("src/main/resources/schema/j2ee_1_4.xsd"),
+                 nameSpace: "j2ee",
+                 packageApi: "org.ironjacamar.shrinkwrap.descriptor.api.j2ee14",
+                 packageImpl:"org.ironjacamar.shrinkwrap.descriptor.impl.j2ee14",
```
Comment (jesperpedersen) [inline embedded/build.gradle]: Ditto

File: embedded/build.gradle [hunk 3]
```diff
@@ -26,8 +29,186 @@ dependencies {
...
+                 generateClasses:"true")
+      descriptor(pathToXsd: project(':common').file("src/main/resources/schema/connector_1_5.xsd"),
+                 nameSpace: "j2ee",
+                 packageApi: "org.ironjacamar.shrinkwrap.descriptor.api.connector15",
+                 packageImpl:"org.ironjacamar.shrinkwrap.descriptor.impl.connector15",
```
Comment (jesperpedersen) [inline embedded/build.gradle]: Ditto

File: embedded/build.gradle [hunk 4]
```diff
@@ -26,8 +29,186 @@ dependencies {
...
+
+  ant.metadataParser(verbose:"false", classpathRef: 'metadata') {
+    path(pathToApi: "${metadataSrcDir}/java",
+         pathToImpl: "${metadataSrcDir}/java",
+         pathToServices: "${metadataSrcDir}/resources/META-INF/services")
+    descriptors() {
+      descriptor(pathToXsd: project(':common').file("src/main/resources/schema/javaee_6.xsd"),
+                 nameSpace: "javaee",
+                 packageApi: "org.ironjacamar.shrinkwrap.descriptor.api.javaee6",
+                 packageImpl:"org.ironjacamar.shrinkwrap.descriptor.impl.javaee6",
```
Comment (jesperpedersen) [inline embedded/build.gradle]: Ditto

File: embedded/build.gradle [hunk 5]
```diff
@@ -26,8 +29,186 @@ dependencies {
...
+                 generateClasses:"true")
+      descriptor(pathToXsd: project(':common').file("src/main/resources/schema/connector_1_6.xsd"),
+                 nameSpace: "javaee",
+                 packageApi: "org.ironjacamar.shrinkwrap.descriptor.api.connector16",
+                 packageImpl:"org.ironjacamar.shrinkwrap.descriptor.impl.connector16",
```
Comment (jesperpedersen) [inline embedded/build.gradle]: Ditto

File: embedded/build.gradle [hunk 6]
```diff
@@ -26,8 +29,186 @@ dependencies {
...
+
+  ant.metadataParser(verbose:"false", classpathRef: 'metadata') {
+    path(pathToApi: "${metadataSrcDir}/java",
+         pathToImpl: "${metadataSrcDir}/java",
+         pathToServices: "${metadataSrcDir}/resources/META-INF/services")
+    descriptors() {
+      descriptor(pathToXsd: project(':common').file("src/main/resources/schema/javaee_7.xsd"),
+                 nameSpace: "javaee",
+                 packageApi: "org.ironjacamar.shrinkwrap.descriptor.api.javaee7",
+                 packageImpl:"org.ironjacamar.shrinkwrap.descriptor.impl.javaee7",
```
Comment (jesperpedersen) [inline embedded/build.gradle]: Ditto

File: embedded/build.gradle [hunk 7]
```diff
@@ -26,8 +29,186 @@ dependencies {
...
+                 generateClasses:"true")
+      descriptor(pathToXsd: project(':common').file("src/main/resources/schema/connector_1_7.xsd"),
+                 nameSpace: "javaee",
+                 packageApi: "org.ironjacamar.shrinkwrap.descriptor.api.connector17",
+                 packageImpl:"org.ironjacamar.shrinkwrap.descriptor.impl.connector17",
```
Comment (jesperpedersen) [inline embedded/build.gradle]: Ditto

**Comment by jesperpedersen** [comment]
The generated APIs doesn't appear in the 'javadoc' target of embedded.

See http://www.ironjacamar.org/doc/api/embedded/1.2/index.html how the 1.2 API looks.

**Comment by vratsel** [comment]
I'll update it. But I don't see any javadocs generated  for any module.
Where is the javadoc target?
Regards,
Vladimir

2015-06-01 13:32 GMT+02:00 Jesper Pedersen <notifications@github.com>:

**Comment by jesperpedersen** [comment]
It is a standard 'javadoc' target from Gradle"
```
gradle javadoc
```

**Comment by vratsel** [comment]
updated

**Comment by jesperpedersen** [comment]
Merged, thanks for your contribution !

**Pull request merged**